pkg/network/ebpf/c/prebuilt/offset-guess.h

```diff
@@ -27,12 +27,20 @@ static const __u8 GUESS_SADDR_FL6 = 12;
 static const __u8 GUESS_DADDR_FL6 = 13;
 static const __u8 GUESS_SPORT_FL6 = 14;
 static const __u8 GUESS_DPORT_FL6 = 15;
+static const __u8 GUESS_SOCKET_SK = 16;
 
 static const __u8 TRACER_STATE_UNINITIALIZED = 0;
 static const __u8 TRACER_STATE_CHECKING = 1;
 static const __u8 TRACER_STATE_CHECKED = 2;
 static const __u8 TRACER_STATE_READY = 3;
 
+// used to disambiguate the source of guess_offsets call
+// useful when guess_offsets is called "simultaneously" from different places
+typedef enum{
+    any,
+    kprobe_sock_getsockopt,
+} caller;
```

Comment:
Can you explain why this is needed? I thought the state machine would prevent any problems related to this. Specifically, you are checking the `status->what` field before calling `guess_offsets` in `kprobe/sock_common_getsockopt`.

Reply:
sorry, I should have clarified this in the comments.

the issue here is that the `getsockopt(2)` syscall used by the offset guessing triggers both `kprobe/tcp_getsockopt` and `kprobe/sock_common_getsockopt` since they belong to the same call stack.

so the caller argument there is to ensure that `calls from kprobe/tcp_getsockopt` will be ignored, and only calls from `kprobe/sock_common_getsockopt` will be handled when `what == GUESS_SOCKET_SK`.
does that make sense?

Reply:
Could you add a check in `kprobe/tcp_getsockopt` to check `what != GUESS_SOCKET_SK` ?

Reply:
yes, that would achieve the same thing. happy to change it if you think that's better

Reply:
I think the intent would be clearer that way.

Reply:
yeah, on a second thought I agree that's better. done.